kratos/processes/from_json_check_result_process.h

```diff
@@ -0,0 +1,921 @@
+    KRATOS_DEFINE_LOCAL_FLAG( ELEMENTS_DATABASE_INITIALIZED );  /// This flag is used in order to check that elements database are initialized
+
+    /// Containers definition
+    typedef ModelPart::NodesContainerType              NodesArrayType;
+    typedef ModelPart::ElementsContainerType        ElementsArrayType;
+
+    /// The node type definiton
+    typedef Node<3> NodeType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Default constructor.
+     * @param rModel The model where the where the simulation is performed
+     * @param ThisParameters The parameters of configuration
+     */
+    FromJSONCheckResultProcess(
+        Model& rModel,
+        Parameters ThisParameters = Parameters(R"({})")
+        );
+
+    /**
+     * @brief Default constructor.
+     * @param rModelPart The model part where the simulation is performed
+     * @param ThisParameters The parameters of configuration
+     */
+    FromJSONCheckResultProcess(
+        ModelPart& rModelPart,
+        Parameters ThisParameters = Parameters(R"({})")
+        );
+
+    /// Destructor.
+    virtual ~FromJSONCheckResultProcess() {}
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    void operator()()
+    {
+        Execute();
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief This method executes the algorithm that looks for neighbour nodes and elements in a  mesh of prismatic elements
+     */
+    void Execute() override;
+
+    /**
+     * @brief This function is designed for being called at the beginning of the computations right after reading the model and the groups
+     */
+    void ExecuteInitialize() override;
+
+    /**
+     * @brief This function will be executed at every time step AFTER performing the solve phase
+     */
+    void ExecuteFinalizeSolutionStep() override;
+
+    /**
+     * @brief This function is designed for being called at the end of the computations
+     */
+    void ExecuteFinalize() override;
+
+    /**
+     * @brief This function is designed for being called after ExecuteInitialize ONCE to verify that the input is correct.
+     */
+    int Check() override;
+
+    /**
+     * @brief This function returns if the result is correct
+     * @return If the result is correct
+     */
+    bool IsCorrectResult()
+    {
+        return this->Is(CORRECT_RESULT);
+    }
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "FromJSONCheckResultProcess";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "FromJSONCheckResultProcess";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+    ///@}
+protected:
```

Comment:
Why `protected`?

Reply:
Because ParticleMechanicsApplication may want to modify the methods once they implement their process in C++